src/articlehistory.py

```diff
@@ -2,31 +2,62 @@
 import requests
 from datetime import datetime
 from src.revision import Revision, URL
-from src.history import format_timestamp,History
+from src.history import format_timestamp, History
+# from revision import Revision, URL
+# from history import format_timestamp, History
+
+# pylint: disable=C0303,R0913,R0914
+
 
-#pylint: disable=C0303,R0913,R0914
 class ArticleHistory(History):
     '''article revision collection class'''
+
     def __init__(self, titles, user=None, keyword=None, tags=None,
-                startyear=None, startmonth=None, startday=None, 
-                starthour=None, startminute=None, startsecond=None, 
-                endyear=None, endmonth=None, endday=None, endhour=None, 
-                endminute=None, endsecond=None):
+                 startyear=None, startmonth=None, startday=None,
+                 starthour=None, startminute=None, startsecond=None,
+                 endyear=None, endmonth=None, endday=None, endhour=None,
+                 endminute=None, endsecond=None):
         super().init_to_none()
         self.init_to_none()
         super().__init__(titles, user, keyword, tags,
-                        startyear, startmonth, startday, 
-                        starthour, startminute, startsecond,
-                        endyear, endmonth, endday, 
-                        endhour, endminute, endsecond)
+                         startyear, startmonth, startday,
+                         starthour, startminute, startsecond,
+                         endyear, endmonth, endday,
+                         endhour, endminute, endsecond)
 
         self.call_wikipedia_api()
-        #filter()
+        self.filter()
 
     def init_to_none(self):
         '''sets up class data members and initalizes to none'''
         self.pageid: int = None
 
+    def filter_by_keyword(self):
+        '''filters list of revisions by keyword'''
+        for rev in self.revisions.copy():
```

Comment:
Not necessary at the moment, but for efficiency's sake it may be better to avoid copying the list going forward - instead of iterating over a copy and removing non-matches from the real list, we could iterate over the real list as is and construct a new list to replace it with, adding the revisions which match the filter as we find them and swapping the new list into the object at the end. Not sure if this would be better, or if this is even the real bottleneck (more likely to be the network requests we make each time we call revision.get_diff).